File: java/src/Ui/DisplayProductDetailsCommand.java
package Ui;

import Products.Catalogue;
import Products.Sku;

class DisplayProductDetailsCommand implements UserCommand {

	private Catalogue catalogue;

	DisplayProductDetailsCommand(Catalogue catalogue) {
		this.catalogue = catalogue;
	}

	public void run(UserInput cmd) {
		String id = cmd.args(1);
		if (id == null || cmd.args.length != 2) {
			System.err.println("ERROR: Usage: d sku");
			return;
		}
		Sku sku = catalogue.lookup(id);
		if (sku == null)
			System.err.println("ERROR: product code " + id + " not found");
		else {
			System.out.printf("%s\t%s\n\n", sku.id, sku.title);
			for (String line : sku.description())
				System.out.println(line);
		}
	}

}

File: dotnet/CatalogueActions/DisplayProductDetailsCommand.cs
namespace CatalogueActions {

using System;
using Products;
using Ui;

class DisplayProductDetailsCommand : UserCommand {

	private Catalogue catalogue;

	internal DisplayProductDetailsCommand(Catalogue catalogue) {
		this.catalogue = catalogue;
	}

	public void run(UserInput cmd) {
		string id = cmd.Args(1);
		if (id == null || cmd.args.Length != 2) {
			Console.WriteLine("ERROR: Usage: d sku");
			return;
		}
		Sku sku = catalogue.lookup(id);
		if (sku == null)
			Console.WriteLine("ERROR: product code " + id + " not found");
		else {
			Console.WriteLine("{0}\t{1}\t£{2,5:F2}", sku.code[2], sku.title, sku.price / 100.0);
			Console.WriteLine();
			foreach (string line in sku.Description())
				Console.WriteLine(line);
		}
	}

}
}